lib/hawkular_monkey_patches.rb

```diff
@@ -13,23 +13,26 @@ def children_domain_controllers(recursive = false)
         def children_domain_hosts(recursive = false)
           children_domain_hosts = []
           ManageIQ::Providers::Hawkular::MiddlewareManager::SUPPORTED_VERSIONS.each do |version|
-            children_by_type("Domain Host #{version}", recursive)
+            children_domain_hosts += children_by_type("Domain Host #{version}", recursive)
+            break unless children_domain_hosts.empty?
```

Comment:
I guess the `break` it's because it can't happen that exist entities with different versions under the same parent. If this is true, wouldn't be a little better to extract the version to query entities only for the resolved version instead of iterating?
...Although really don't like the idea of extracting from strings something that should be a field.

Reply:
Yes, that's right.  It's a slight optimization.  I don't want to get into string magic here, perhaps it can be further optimized later but for now I think it's fine.